src/pages/gen2/start/quickstart/nextjs-app-router-server-components/index.mdx

```diff
@@ -308,9 +349,12 @@ After creating a todo, `revalidatePath` is run to clear the Next.js cache for th
 
 import { revalidatePath } from "next/cache";
 
-import { cookiesClient } from "@/utils/amplify-utils";
+import { AuthGetCurrentUserServer, cookiesClient } from "@/utils/amplify-utils";
+
+import Logout from "@/components/Logout";
 
 async function App() {
+  const user = await AuthGetCurrentUserServer();
```

Comment:
How would this work with federated auth since the code exchange happens in the client side? If a customer has already logged in Via googele, and this app is loaded again on the sever side, wouldn't user be null since no cookies would be set yet? I believe the Logout component may show on initial load after federated auth because the user will be missing even though technically the customer would be logged in.

Reply:
Good points, and I've set up a meeting to discuss.

Reply:
Per our conversation, adding a section to have callback urls redirect to `/login` for Social Providers.
<img width="1090" alt="Screenshot 2024-01-30 at 10 24 19 AM" src="https://github.com/aws-amplify/docs/assets/21967/69b54def-9d98-45d3-93fb-8ecc2a6aa6a7">